<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Goal" data-toc-modified-id="Goal-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Goal</a></span></li><li><span><a href="#Var" data-toc-modified-id="Var-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Var</a></span></li><li><span><a href="#Init" data-toc-modified-id="Init-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Init</a></span></li><li><span><a href="#BioProjects" data-toc-modified-id="BioProjects-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>BioProjects</a></span></li><li><span><a href="#LLMGQC" data-toc-modified-id="LLMGQC-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>LLMGQC</a></span><ul class="toc-item"><li><span><a href="#PRJEB11755" data-toc-modified-id="PRJEB11755-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>PRJEB11755</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.1.1"><span class="toc-item-num">5.1.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Problem-samples" data-toc-modified-id="Problem-samples-5.1.2"><span class="toc-item-num">5.1.2&nbsp;&nbsp;</span>Problem samples</a></span></li><li><span><a href="#Cutting-loses" data-toc-modified-id="Cutting-loses-5.1.3"><span class="toc-item-num">5.1.3&nbsp;&nbsp;</span>Cutting loses</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.1.4"><span class="toc-item-num">5.1.4&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB20308" data-toc-modified-id="PRJEB20308-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>PRJEB20308</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.2.1"><span class="toc-item-num">5.2.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Cutting-loses" data-toc-modified-id="Cutting-loses-5.2.2"><span class="toc-item-num">5.2.2&nbsp;&nbsp;</span>Cutting loses</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.2.3"><span class="toc-item-num">5.2.3&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB21624" data-toc-modified-id="PRJEB21624-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>PRJEB21624</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.3.1"><span class="toc-item-num">5.3.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.3.2"><span class="toc-item-num">5.3.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB29346" data-toc-modified-id="PRJEB29346-5.4"><span class="toc-item-num">5.4&nbsp;&nbsp;</span>PRJEB29346</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.4.1"><span class="toc-item-num">5.4.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Problem-samples" data-toc-modified-id="Problem-samples-5.4.2"><span class="toc-item-num">5.4.2&nbsp;&nbsp;</span>Problem samples</a></span></li><li><span><a href="#Cutting-loses" data-toc-modified-id="Cutting-loses-5.4.3"><span class="toc-item-num">5.4.3&nbsp;&nbsp;</span>Cutting loses</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.4.4"><span class="toc-item-num">5.4.4&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA316560" data-toc-modified-id="PRJNA316560-5.5"><span class="toc-item-num">5.5&nbsp;&nbsp;</span>PRJNA316560</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.5.1"><span class="toc-item-num">5.5.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.5.2"><span class="toc-item-num">5.5.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA316570" data-toc-modified-id="PRJNA316570-5.6"><span class="toc-item-num">5.6&nbsp;&nbsp;</span>PRJNA316570</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.6.1"><span class="toc-item-num">5.6.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.6.2"><span class="toc-item-num">5.6.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA336354" data-toc-modified-id="PRJNA336354-5.7"><span class="toc-item-num">5.7&nbsp;&nbsp;</span>PRJNA336354</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.7.1"><span class="toc-item-num">5.7.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.7.2"><span class="toc-item-num">5.7.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA353155" data-toc-modified-id="PRJNA353155-5.8"><span class="toc-item-num">5.8&nbsp;&nbsp;</span>PRJNA353155</a></span><ul class="toc-item"><li><span><a href="#SKIPPING" data-toc-modified-id="SKIPPING-5.8.1"><span class="toc-item-num">5.8.1&nbsp;&nbsp;</span>SKIPPING</a></span></li></ul></li><li><span><a href="#PRJNA381379" data-toc-modified-id="PRJNA381379-5.9"><span class="toc-item-num">5.9&nbsp;&nbsp;</span>PRJNA381379</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.9.1"><span class="toc-item-num">5.9.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.9.2"><span class="toc-item-num">5.9.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA390686" data-toc-modified-id="PRJNA390686-5.10"><span class="toc-item-num">5.10&nbsp;&nbsp;</span>PRJNA390686</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.10.1"><span class="toc-item-num">5.10.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.10.2"><span class="toc-item-num">5.10.2&nbsp;&nbsp;</span>Summary</a></span><ul class="toc-item"><li><span><a href="#Notes" data-toc-modified-id="Notes-5.10.2.1"><span class="toc-item-num">5.10.2.1&nbsp;&nbsp;</span>Notes</a></span></li></ul></li></ul></li><li><span><a href="#PRJNA417359" data-toc-modified-id="PRJNA417359-5.11"><span class="toc-item-num">5.11&nbsp;&nbsp;</span>PRJNA417359</a></span><ul class="toc-item"><li><span><a href="#Subsample" data-toc-modified-id="Subsample-5.11.1"><span class="toc-item-num">5.11.1&nbsp;&nbsp;</span>Subsample</a></span></li><li><span><a href="#Config" data-toc-modified-id="Config-5.11.2"><span class="toc-item-num">5.11.2&nbsp;&nbsp;</span>Config</a></span></li><li><span><a href="#Run" data-toc-modified-id="Run-5.11.3"><span class="toc-item-num">5.11.3&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.11.4"><span class="toc-item-num">5.11.4&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA476660" data-toc-modified-id="PRJNA476660-5.12"><span class="toc-item-num">5.12&nbsp;&nbsp;</span>PRJNA476660</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.12.1"><span class="toc-item-num">5.12.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.12.2"><span class="toc-item-num">5.12.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA485217" data-toc-modified-id="PRJNA485217-5.13"><span class="toc-item-num">5.13&nbsp;&nbsp;</span>PRJNA485217</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.13.1"><span class="toc-item-num">5.13.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.13.2"><span class="toc-item-num">5.13.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJNA532626" data-toc-modified-id="PRJNA532626-5.14"><span class="toc-item-num">5.14&nbsp;&nbsp;</span>PRJNA532626</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.14.1"><span class="toc-item-num">5.14.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.14.2"><span class="toc-item-num">5.14.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB22765" data-toc-modified-id="PRJEB22765-5.15"><span class="toc-item-num">5.15&nbsp;&nbsp;</span>PRJEB22765</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.15.1"><span class="toc-item-num">5.15.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.15.2"><span class="toc-item-num">5.15.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB23642" data-toc-modified-id="PRJEB23642-5.16"><span class="toc-item-num">5.16&nbsp;&nbsp;</span>PRJEB23642</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.16.1"><span class="toc-item-num">5.16.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.16.2"><span class="toc-item-num">5.16.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li><li><span><a href="#PRJEB9357" data-toc-modified-id="PRJEB9357-5.17"><span class="toc-item-num">5.17&nbsp;&nbsp;</span>PRJEB9357</a></span><ul class="toc-item"><li><span><a href="#Run" data-toc-modified-id="Run-5.17.1"><span class="toc-item-num">5.17.1&nbsp;&nbsp;</span>Run</a></span></li><li><span><a href="#Summary" data-toc-modified-id="Summary-5.17.2"><span class="toc-item-num">5.17.2&nbsp;&nbsp;</span>Summary</a></span></li></ul></li></ul></li><li><span><a href="#sessionInfo" data-toc-modified-id="sessionInfo-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>sessionInfo</a></span></li></ul></div>

# Goal

* Run LLMGQC pipeline on all BioProject samples sets

# Var

In [2]:
work_dir = '/ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/'
pipeline_dir = '/ebio/abt3_projects/Georg_animal_feces/bin/llmgqc/'

# Init

In [12]:
library(dplyr)
library(tidyr)
library(ggplot2)
set.seed(1831)
source('/ebio/abt3_projects/Georg_animal_feces/code/misc_r_functions/init.R')

# BioProjects

In [4]:
samples_files = list.files(work_dir, 'samples.txt', full.names=TRUE, recursive=TRUE)
samples_files %>% length
samples_files

# LLMGQC

* Running on each bioproject separately

## PRJEB11755

* host = pig
* num samples = 295
* avg. read length = ~100
* pipeline dir  = `/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc`

In [28]:
cat_file(file.path(work_dir, 'PRJEB11755', 'LLMGQC', 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB11755/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB11755/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote (SRA) sa

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/Georg_animal_feces/bin/llmgqc
$ screen -L -S llmgqc-ga ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB11755/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB11755/SGE_log 20
```

### Problem samples

* Having a problem downloading some samples

In [ ]:
projectID = 'PRJEB11755'

In [12]:
P = file.path(work_dir, projectID, 'LLMGQC', 'final')
good_samps = list.files(P, 'R1_final.fq.gz', recursive=TRUE, full.names=TRUE)
good_samps %>% length %>% print
good_samps = good_samps %>% dirname %>% basename
good_samps %>% head

[1] 146


[1] "SAMEA3663006" "SAMEA3663007" "SAMEA3663010" "SAMEA3663013" "SAMEA3663015"
[6] "SAMEA3663017"

In [19]:
F = file.path(work_dir, projectID, 'LLMGQC', 'samples.txt')
df_samps = read.delim(F, sep='\t') %>%
    mutate(qc_finished = Sample %in% good_samps)
df_samps %>% .$qc_finished %>% summary %>% print
df_samps %>% dfhead

   Mode   FALSE    TRUE 
logical     149     146 
[1] 295  42


Sample       Run        Lane Remote     spots    bases      avgLength size_MB
1 SAMEA3663295 ERR1135468 1    ERR1135468 24873864 4875031192 195       3574   
2 SAMEA3663296 ERR1135469 1    ERR1135469 29496968 5791369887 196       4228   
3 SAMEA3663297 ERR1135470 1    ERR1135470 26828175 5160184344 192       3957   
  BioProject bioproject ⋯ SRA_accession Title  Animal First_author Year
1 PRJEB11755 PRJEB11755 ⋯ ERS970444     ZXZ-5B Pig    Xiao         2016
2 PRJEB11755 PRJEB11755 ⋯ ERS970445     ZXZ-6B Pig    Xiao         2016
3 PRJEB11755 PRJEB11755 ⋯ ERS970446     ZXZ-7B Pig    Xiao         2016
  DOI                                         Sequencer Read_length Comments
1 https://doi.org/10.1038/nmicrobiol.2016.161 Hiseq2000 100         NA      
2 https://doi.org/10.1038/nmicrobiol.2016.161 Hiseq2000 100         NA      
3 https://doi.org/10.1038/nmicrobiol.2016.161 Hiseq2000 100         NA      
  qc_finished
1 TRUE       
2 TRUE       
3 TRUE

In [24]:
df_samps %>%
    filter(qc_finished == TRUE) %>%
    .$Title %>% sort

[1] BHZ-10B   BHZ-12B   BHZ-1B    BHZ-2B    BHZ-4B    BHZ-5B    BHZ-8B   
  [8] BMZ-11B   BMZ-14B   BMZ-15B   BMZ-16B   BMZ-17B   BMZ-3B    BMZ-6B   
 [15] BMZ-9B    DB-510B   DB-512B   DB-530B   DB-531B   DB-537B   EYZ-120B 
 [22] EYZ-160Ba EYZ-192Bb EYZ-353B  EYZ-362B  EYZ-368B  EYZ-378B  EYZ-468Ba
 [29] EYZ-542B  EYZ-566Bb EYZ-592Ba EYZ-592Bb EYZ-652Bb PIG_111   PIG_112  
 [36] PIG_114   PIG_115   PIG_118   PIG_119   PIG_121   PIG_122   PIG_124  
 [43] PIG_130   PIG_131   PIG_132   PIG_134   PIG_135   PIG_136   PIG_141  
 [50] PIG_142   PIG_144   PIG_146   PIG_153   PIG_161   PIG_164   PIG_168  
 [57] PIG_170   PIG_172   PIG_174   PIG_179   PIG_181   PIG_183   PIG_184  
 [64] PIG_185   PIG_186   PIG_190   PIG_192   PIG_193   PIG_197   PIG_202  
 [71] PIG_205   PIG_206   PIG_207   PIG-002   PIG-003   PIG-006   PIG-009  
 [78] PIG-012   PIG-015   PIG-020   PIG-022   PIG-023   PIG-024   PIG-025  
 [85] PIG-027   PIG-035   PIG-037   PIG-038   PIG-039   PIG-046   PIG-047  
 [92] PIG-048   PIG-049   PIG-052   PIG-054   PIG-055   PIG-056   PIG-059  
 [99] PIG-060   PIG-062   PIG-063   PIG-066   PIG-067   PIG-071   PIG-073  
[106] PIG-074   PIG-076   PIG-078   PIG-080   PIG-081   PIG-086   PIG-087  
[113] PIG-088   PIG-090   PIG-091   PIG-095   PIG-096   PIG-097   PIG-098  
[120] PIG-099   PIG-101   PIG-102   PIG-103   PIG-105   PIG-109   SYZ-202B 
[127] SYZ-304B  SYZ-388B  SYZ-423B  SYZ-426B  SYZ-430B  SYZ-441B  SYZ-442B 
[134] SYZ-448B  SYZ-477B  SYZ-481B  SYZ-495B  SYZ-636B  SYZ-639B  SYZ-A002B
[141] ZXZ-10Ba  ZXZ-3B    ZXZ-5B    ZXZ-6B    ZXZ-7B    ZXZ-9B   
295 Levels: BHZ-10B BHZ-11B BHZ-12B BHZ-1B BHZ-2B BHZ-3B BHZ-4B ... ZXZ-9B

In [25]:
df_samps %>%
    filter(qc_finished == FALSE) %>%
    .$Title %>% sort

[1] BHZ-11B   BHZ-3B    BHZ-6B    BHZ-7B    BHZ-9B    BMZ-10B   BMZ-12B  
  [8] BMZ-13B   BMZ-1B    BMZ-2B    BMZ-4B    BMZ-5B    BMZ-7B    BMZ-8B   
 [15] DB-11B    DB-511B   DB-515B   DB-517B   DB-529B   EYZ-160Bb EYZ-183B 
 [22] EYZ-192Ba EYZ-254B  EYZ-381B  EYZ-468Bb EYZ-566Ba EYZ-568B  EYZ-572Ba
 [29] EYZ-572Bb EYZ-611B  EYZ-615B  EYZ-652Ba PIG_113   PIG_116   PIG_117  
 [36] PIG_120   PIG_123   PIG_125   PIG_126   PIG_127   PIG_128   PIG_129  
 [43] PIG_133   PIG_137   PIG_138   PIG_139   PIG_140   PIG_143   PIG_145  
 [50] PIG_147   PIG_148   PIG_149   PIG_150   PIG_151   PIG_152   PIG_154  
 [57] PIG_155   PIG_156   PIG_157   PIG_158   PIG_159   PIG_160   PIG_162  
 [64] PIG_163   PIG_165   PIG_166   PIG_167   PIG_169   PIG_171   PIG_173  
 [71] PIG_175   PIG_176   PIG_177   PIG_178   PIG_180   PIG_182   PIG_187  
 [78] PIG_188   PIG_189   PIG_191   PIG_194   PIG_195   PIG_196   PIG_198  
 [85] PIG_199   PIG_200   PIG_201   PIG_203   PIG_204   PIG_208   PIG_209  
 [92] PIG_210   PIG-001   PIG-004   PIG-005   PIG-007   PIG-008   PIG-011  
 [99] PIG-014   PIG-016   PIG-017   PIG-018   PIG-019   PIG-026   PIG-028  
[106] PIG-029   PIG-030   PIG-031   PIG-032   PIG-033   PIG-034   PIG-036  
[113] PIG-041   PIG-042   PIG-043   PIG-044   PIG-045   PIG-053   PIG-057  
[120] PIG-058   PIG-061   PIG-064   PIG-065   PIG-068   PIG-069   PIG-075  
[127] PIG-077   PIG-079   PIG-082   PIG-083   PIG-084   PIG-085   PIG-092  
[134] PIG-093   PIG-094   PIG-104   PIG-106   PIG-107   PIG-108   PIG-110  
[141] SYZ-428B  SYZ-533B  SYZ-631B  SYZ-635B  ZXZ-10Bb  ZXZ-1B    ZXZ-2B   
[148] ZXZ-4B    ZXZ-8B   
295 Levels: BHZ-10B BHZ-11B BHZ-12B BHZ-1B BHZ-2B BHZ-3B BHZ-4B ... ZXZ-9B

### Cutting loses

* finishing current downloaded set

In [26]:
F = file.path(work_dir, 'PRJEB11755', 'samples_success.txt')
df_samps %>%
    filter(qc_finished == TRUE) %>%
    write.table(F, sep='\t', quote=FALSE, row.names=FALSE)
cat('File written:', F, '\n')

File written: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB11755/samples_success.txt 


### Summary

In [38]:
F = file.path(work_dir, 'PRJEB11755', 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 292  21


Sample                                      Filename      
1 SAMEA3663006 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA3663006 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA3663007 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 25857722       
2 Conventional base calls Sanger / Illumina 1.9 25857722       
3 Conventional base calls Sanger / Illumina 1.9 27881309       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-96           40  
2 0                                 90-93           40  
3 0                                 90-95           44  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 96.68032                      95.92704            ⋯ pass                     
2 96.26200                      92.93400            ⋯ pass                     
3 96.99528                      94.91033            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 fail                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 pass                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [40]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   : 8763549   Min.   :90.97      
 1st Qu.:22726842   1st Qu.:94.89      
 Median :27747114   Median :95.83      
 Mean   :26647479   Mean   :96.63      
 3rd Qu.:31592658   3rd Qu.:99.67      
 Max.   :42467986   Max.   :99.87      

## PRJEB20308

* host = dog
* n = 3096
* avg. read length = 250

In [62]:
projectID = 'PRJEB20308'

In [63]:
cat_file(file.path(work_dir, projectID, 'LLMGQC', 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
#samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB20308/LLMGQC/samples.txt
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB20308/LLMGQC/samples_success.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB20308/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg.,

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/Georg_animal_feces/bin/llmgqc
$ screen -L -S llmgqc-PRJEB20308 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB20308/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB20308/LLMGQC/SGE_log/ 20
```

### Cutting loses

In [58]:
projectID = 'PRJEB20308'

In [64]:
P = file.path(work_dir, projectID, 'LLMGQC', 'final')
good_samps = list.files(P, 'R1_final.fq.gz', recursive=TRUE, full.names=TRUE)
good_samps %>% length %>% print
good_samps = good_samps %>% dirname %>% basename
good_samps %>% head

[1] 122


[1] "Ash_p1"       "Ash_p2"       "Avalanche_p1" "Avalanche_p2" "Bandit_p1"   
[6] "Bandit_p2"

In [56]:
F = file.path(work_dir, projectID, 'samples.txt')
df_samps = read.delim(F, sep='\t') %>%
    mutate(qc_finished = Sample %in% good_samps)
df_samps %>% .$qc_finished %>% summary %>% print
df_samps %>% dfhead

   Mode   FALSE    TRUE 
logical     168    2928 
[1] 3096   42


Sample Run        Lane Remote     spots   bases     avgLength size_MB
1 Ash_p1 ERR1915336 1    ERR1915336 1998894 503721288 252       206    
2 Ash_p1 ERR1915337 1    ERR1915337 1996246 503053992 252       207    
3 Ash_p1 ERR1915338 1    ERR1915338 1968665 496103580 252       206    
  BioProject bioproject ⋯ SRA_accession Title                                 
1 PRJEB20308 PRJEB20308 ⋯ NA            MIMARKS Specimen sample from bacterium
2 PRJEB20308 PRJEB20308 ⋯ NA            MIMARKS Specimen sample from bacterium
3 PRJEB20308 PRJEB20308 ⋯ NA            MIMARKS Specimen sample from bacterium
  Animal First_author Year DOI                                       Sequencer
1 Dog    Coelho       2018 https://doi.org/10.1186/s40168-018-0450-3 HiSeq2500
2 Dog    Coelho       2018 https://doi.org/10.1186/s40168-018-0450-3 HiSeq2500
3 Dog    Coelho       2018 https://doi.org/10.1186/s40168-018-0450-3 HiSeq2500
  Read_length Comments qc_finished
1 125         NA       TRUE       
2 125         NA       TRUE       
3 125         NA       TRUE

In [57]:
F = file.path(work_dir, projectID, 'samples_success.txt')
df_samps %>%
    filter(qc_finished == TRUE) %>%
    write.table(F, sep='\t', quote=FALSE, row.names=FALSE)
cat('File written:', F, '\n')

File written: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB20308/samples_success.txt 


### Summary

In [65]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 241  21


Sample                                Filename       File.type              
1 Ash_p1 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
2 Ash_p1 | fastqc_per_sample | R2_final R2_final.fq.gz Conventional base calls
3 Ash_p2 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
  Encoding              Total.Sequences Sequences.flagged.as.poor.quality
1 Sanger / Illumina 1.9 49060464        0                                
2 Sanger / Illumina 1.9 49060464        0                                
3 Sanger / Illumina 1.9 39179808        0                                
  Sequence.length X.GC total_deduplicated_percentage avg_sequence_length ⋯
1 90-126          44   51.14093                      125.4655            ⋯
2 90-126          44   49.88428                      125.3960            ⋯
3 90-125          46   62.39226                      124.2897            ⋯
  per_base_sequence_quality per_tile_sequence_quality
1 pass                      pass                     
2 pass                      pass                     
3 pass                      pass                     
  per_sequence_quality_scores per_base_sequence_content per_sequence_gc_content
1 pass                        fail                      pass                   
2 pass                        fail                      pass                   
3 pass                        warn                      warn                   
  per_base_n_content sequence_length_distribution sequence_duplication_levels
1 pass               warn                         warn                       
2 pass               warn                         fail                       
3 pass               warn                         warn                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [66]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences     avg_sequence_length
 Min.   : 29485375   Min.   :123.5      
 1st Qu.: 43809476   1st Qu.:124.2      
 Median : 49567232   Median :124.6      
 Mean   : 53855723   Mean   :124.7      
 3rd Qu.: 59929862   3rd Qu.:125.4      
 Max.   :104069583   Max.   :125.7      

## PRJEB21624

* host = cattle
* n = 43
* avg. read length = 300

In [67]:
projectID = 'PRJEB21624'

In [68]:
cat_file(file.path(work_dir, projectID, 'LLMGQC', 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB21624/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB21624/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote (

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJEB21624 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB21624/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB21624/LLMGQC/SGE_log 20
```

### Summary

In [69]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 81 21


Sample                                        Filename      
1 SAMEA104153961 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA104153961 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA104153962 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 66215805       
2 Conventional base calls Sanger / Illumina 1.9 24751756       
3 Conventional base calls Sanger / Illumina 1.9 61417360       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-150          43  
2 0                                 90-150          42  
3 0                                 90-150          44  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 94.12778                      149.5320            ⋯ pass                     
2 95.87157                      148.6961            ⋯ pass                     
3 92.89022                      149.5553            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      fail                    pass              
2 pass                      fail                    pass              
3 pass                      fail                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [70]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences     avg_sequence_length
 Min.   : 50003078   Min.   :148.9      
 1st Qu.: 54376472   1st Qu.:149.7      
 Median : 59507399   Median :149.7      
 Mean   : 62886361   Mean   :149.7      
 3rd Qu.: 67105390   3rd Qu.:149.8      
 Max.   :117403834   Max.   :149.8      

## PRJEB29346

* host = cod
* n = 57
* avg. read length = 250

In [79]:
projectID = 'PRJEB29346'

In [53]:
cat_file(file.path(work_dir, projectID, 'LLMGQC', 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB29346/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB29346/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote (

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/vadinCA11/bin/llmgqc
$ screen -L -S llmgqc-PRJEB29346 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB29346/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB29346/LLMGQC/ 20
```

### Problem samples

In [82]:
P = file.path(work_dir, projectID, 'LLMGQC', 'final')
good_samps = list.files(P, 'R1_final.fq.gz', recursive=TRUE, full.names=TRUE)
good_samps %>% length %>% print
good_samps = good_samps %>% dirname %>% basename
good_samps %>% head

[1] 56


[1] "SAMEA5043640" "SAMEA5043641" "SAMEA5043642" "SAMEA5043643" "SAMEA5043644"
[6] "SAMEA5043645"

In [84]:
F = file.path(work_dir, projectID, 'samples.txt')
df_samps = read.delim(F, sep='\t') %>%
    mutate(qc_finished = Sample %in% good_samps)

df_samps %>% .$qc_finished %>% summary %>% print
df_samps %>% dfhead

   Mode   FALSE    TRUE 
logical       1      56 
[1] 57 42


Sample       Run        Lane Remote     spots   bases     avgLength size_MB
1 SAMEA5043640 ERR2856134 1    ERR2856134 3245792 817939584 252       270    
2 SAMEA5043641 ERR2856135 1    ERR2856135 3794750 956277000 252       317    
3 SAMEA5043642 ERR2856136 1    ERR2856136 3843198 968485896 252       321    
  BioProject bioproject ⋯ SRA_accession Title          Animal First_author Year
1 PRJEB29346 PRJEB29346 ⋯ ERS2855374    L_01_r_01_l_01 Cod    Riiser       2019
2 PRJEB29346 PRJEB29346 ⋯ ERS2855375    L_01_r_02_l_01 Cod    Riiser       2019
3 PRJEB29346 PRJEB29346 ⋯ ERS2855376    L_01_r_02_l_02 Cod    Riiser       2019
  DOI                                     Sequencer Read_length Comments
1 https://doi.org/10.1111/1462-2920.14652 HiSeq2500 125         NA      
2 https://doi.org/10.1111/1462-2920.14652 HiSeq2500 125         NA      
3 https://doi.org/10.1111/1462-2920.14652 HiSeq2500 125         NA      
  qc_finished
1 TRUE       
2 TRUE       
3 TRUE

### Cutting loses

In [85]:
F = file.path(work_dir, projectID, 'LLMGQC', 'samples_success.txt')
df_samps %>%
    filter(qc_finished == TRUE) %>%
    write.table(F, sep='\t', quote=FALSE, row.names=FALSE)
cat('File written:', F, '\n')

File written: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB29346/LLMGQC/samples_success.txt 


### Summary

In [86]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 112  21


Sample                                      Filename      
1 SAMEA5043640 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA5043640 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA5043641 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 3026546        
2 Conventional base calls Sanger / Illumina 1.9 3026546        
3 Conventional base calls Sanger / Illumina 1.9 3536098        
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-126          44  
2 0                                 90-126          44  
3 0                                 90-126          43  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 96.35073                      125.0893            ⋯ pass                     
2 96.27242                      124.8990            ⋯ pass                     
3 96.10727                      125.1376            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 warn                      pass                       
2 pass                      pass                       
3 warn                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 warn                      pass                    pass              
2 warn                      pass                    pass              
3 warn                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [87]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences   avg_sequence_length
 Min.   :2177605   Min.   :123.9      
 1st Qu.:2832577   1st Qu.:125.0      
 Median :3113354   Median :125.0      
 Mean   :3162384   Mean   :125.0      
 3rd Qu.:3535400   3rd Qu.:125.2      
 Max.   :4734017   Max.   :125.4      

## PRJNA316560

* host = goose
* n = 2
* avg. read length = 250

In [71]:
projectID = 'PRJNA316560'

In [74]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA316560/LLMGQC 


In [75]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316560/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316560/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJNA316560 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316560/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316560/LLMGQC/SGE_log 20
```

### Summary

In [76]:
projectID = 'PRJNA316560'

In [77]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1]  4 20


Sample                                               Filename      
1 artificially_reared_1 | fastqc_per_sample | R1_final R1_final.fq.gz
2 artificially_reared_1 | fastqc_per_sample | R2_final R2_final.fq.gz
3 artificially_reared_2 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 14682965       
2 Conventional base calls Sanger / Illumina 1.9 14682965       
3 Conventional base calls Sanger / Illumina 1.9 20768849       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-126          43  
2 0                                 90-126          42  
3 0                                 90-126          44  
  total_deduplicated_percentage avg_sequence_length basic_statistics
1 91.10658                      123.5055            pass            
2 91.48249                      124.8729            pass            
3 94.94694                      123.2767            pass            
  per_base_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 warn                      pass                    pass              
3 warn                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [78]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   :14682965   Min.   :123.3      
 1st Qu.:16204436   1st Qu.:123.3      
 Median :17725907   Median :123.4      
 Mean   :17725907   Mean   :123.4      
 3rd Qu.:19247378   3rd Qu.:123.4      
 Max.   :20768849   Max.   :123.5      

## PRJNA316570

* host = goose
* n = 2
* avg. read length = 250

In [89]:
projectID = 'PRJNA316570'

In [90]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA316570/LLMGQC 


In [91]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316570/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316570/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJNA316570 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316570/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA316570/LLMGQC/SGE_log 20
```

### Summary

In [92]:
projectID = 'PRJNA316570'

In [93]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1]  4 20


Sample                                       Filename      
1 wild_sample_1 | fastqc_per_sample | R1_final R1_final.fq.gz
2 wild_sample_1 | fastqc_per_sample | R2_final R2_final.fq.gz
3 wild_sample_2 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 18167849       
2 Conventional base calls Sanger / Illumina 1.9 18167849       
3 Conventional base calls Sanger / Illumina 1.9 16891888       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-126          41  
2 0                                 90-126          41  
3 0                                 90-126          42  
  total_deduplicated_percentage avg_sequence_length basic_statistics
1 90.24708                      123.6654            pass            
2 90.93926                      124.8679            pass            
3 90.47841                      123.3605            pass            
  per_base_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 warn                      pass                    pass              
3 pass                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [94]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   :16891888   Min.   :123.4      
 1st Qu.:17210878   1st Qu.:123.4      
 Median :17529868   Median :123.5      
 Mean   :17529868   Mean   :123.5      
 3rd Qu.:17848859   3rd Qu.:123.6      
 Max.   :18167849   Max.   :123.7      

## PRJNA336354

* host = woodrat
* n = 15
* avg. read length = 201

In [95]:
projectID = 'PRJNA336354'

In [96]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Directory already exists: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA336354/LLMGQC 


In [97]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA336354/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA336354/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgp-PRJNA336354 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA336354/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA336354/LLMGQC/SGE_log/ 20
```

### Summary

In [98]:
projectID = 'PRJNA336354'

In [99]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 30 21


Sample                                                 Filename      
1 C._urophasianus_2_cecum | fastqc_per_sample | R1_final R1_final.fq.gz
2 C._urophasianus_2_cecum | fastqc_per_sample | R2_final R2_final.fq.gz
3 C._urophasianus_3_cecum | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 3782779        
2 Conventional base calls Sanger / Illumina 1.9 3782779        
3 Conventional base calls Sanger / Illumina 1.9 1179255        
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-101          46  
2 0                                 90-101          47  
3 0                                 90-101          42  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 97.51923                      100.5174            ⋯ pass                     
2 97.33972                      100.7532            ⋯ pass                     
3 98.05388                      100.5258            ⋯ pass                     
  per_sequence_quality_scores per_base_sequence_content per_sequence_gc_content
1 pass                        warn                      fail                   
2 pass                        warn                      fail                   
3 pass                        warn                      fail                   
  per_base_n_content sequence_length_distribution sequence_duplication_levels
1 pass               warn                         pass                       
2 pass               warn                         pass                       
3 pass               warn                         pass                       
  overrepresented_sequences adapter_content per_tile_sequence_quality
1 pass                      pass                                     
2 pass                      pass                                     
3 pass                      pass

In [100]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences   avg_sequence_length
 Min.   :1129979   Min.   :100.5      
 1st Qu.:2120016   1st Qu.:100.5      
 Median :3706963   Median :100.5      
 Mean   :3731855   Mean   :100.5      
 3rd Qu.:5021424   3rd Qu.:100.6      
 Max.   :6942214   Max.   :100.7      

## PRJNA353155

* host = carp
* n = 1
* avg. read length = 150

### SKIPPING

## PRJNA381379

* host = Kakapo
* n = 6 (3 adults & 3 chicks)
* avg. read length = 300
* seq. tech. = MiSeq

In [101]:
projectID = 'PRJNA381379'

In [102]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Directory already exists: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA381379/LLMGQC 


In [104]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA381379/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA381379/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJNA381379 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA381379/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA381379/LLMGQC/SGE_log/ 20
```

### Summary

In [110]:
projectID = 'PRJNA381379'

In [111]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 12 20


Sample                                 Filename       File.type              
1 Adult_1 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
2 Adult_1 | fastqc_per_sample | R2_final R2_final.fq.gz Conventional base calls
3 Adult_2 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
  Encoding              Total.Sequences Sequences.flagged.as.poor.quality
1 Sanger / Illumina 1.9 2477917         0                                
2 Sanger / Illumina 1.9 2477917         0                                
3 Sanger / Illumina 1.9 2475317         0                                
  Sequence.length X.GC total_deduplicated_percentage avg_sequence_length
1 90-251          50   70.90194                      165.1839           
2 90-251          51   76.26464                      164.5711           
3 90-251          52   71.13014                      173.3867           
  basic_statistics per_base_sequence_quality per_sequence_quality_scores
1 pass             pass                      pass                       
2 pass             warn                      pass                       
3 pass             pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 fail                      warn                    pass              
2 fail                      warn                    pass              
3 fail                      warn                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [112]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences   avg_sequence_length
 Min.   :1906620   Min.   :165.2      
 1st Qu.:2357625   1st Qu.:167.7      
 Median :2476617   Median :171.4      
 Mean   :2503786   Mean   :176.6      
 3rd Qu.:2563651   3rd Qu.:187.3      
 Max.   :3252238   Max.   :192.7      

## PRJNA390686

* host = Mouse
* n = 39
* avg. read length = 150

In [5]:
projectID = 'PRJNA390686'

In [6]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA390686/LLMGQC 


In [8]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA390686/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA390686/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJNA390686 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA390686/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA390686/LLMGQC/SGE_log/ 20
```

### Summary

In [66]:
projectID = 'PRJNA390686'

#### Notes

* Samples failed to download; skipping this dataset

## PRJNA417359

* host = chicken
* n = 465
* avg. read length = 280

In [28]:
projectID = 'PRJNA417359'

In [29]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Directory already exists: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA417359/LLMGQC 


### Subsample 

In [35]:
F = file.path(work_dir, projectID, 'samples.txt')
samps = read.delim(F, sep='\t')
samps %>% dfhead

[1] 465  41


Sample     Run        Lane Remote     spots    bases      avgLength size_MB
1 CBJMR04201 SRR8840694 1    SRR8840694  9945201 2672474479 268       1170   
2 DGYFJ06201 SRR6323428 1    SRR6323428 17489115 4334964047 247       1553   
3 DSTMI04001 SRR6323432 1    SRR6323432 11041309 2722146872 246        972   
  BioProject  bioproject  ⋯
1 PRJNA417359 PRJNA417359 ⋯
2 PRJNA417359 PRJNA417359 ⋯
3 PRJNA417359 PRJNA417359 ⋯
  Name                                                                                 
1 Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
2 Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
3 Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
  SRA_accession Title                                                         
1 NA            Metagenome or environmental sample from chicken gut metagenome
2 NA            Metagenome or environmental sample from chicken gut metagenome
3 NA            Metagenome or environmental sample from chicken gut metagenome
  Animal  First_author Year DOI                                      
1 Chicken Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
2 Chicken Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
3 Chicken Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
  Sequencer           Read_length Comments
1 HiSeq2500, HiSeqX10 NA          NA      
2 HiSeq2500, HiSeqX10 NA          NA      
3 HiSeq2500, HiSeqX10 NA          NA

In [36]:
samps %>% 
    dplyr::select(spots, bases, avgLength, size_MB) %>%
    summary

     spots              bases             avgLength        size_MB    
 Min.   : 1050368   Min.   :2.726e+08   Min.   :235.0   Min.   : 118  
 1st Qu.: 8929103   1st Qu.:2.483e+09   1st Qu.:247.0   1st Qu.:1011  
 Median :12578365   Median :3.287e+09   Median :280.0   Median :1285  
 Mean   :12798261   Mean   :3.512e+09   Mean   :273.4   Mean   :1352  
 3rd Qu.:16846141   3rd Qu.:4.559e+09   3rd Qu.:288.0   3rd Qu.:1695  
 Max.   :40496042   Max.   :1.186e+10   Max.   :294.0   Max.   :4373  

In [37]:
# subsampling
samps = samps %>%
    filter(spots >= 10000000) %>%
    sample_n(130)

samps %>% dfhead

[1] 130  41


Sample     Run        Lane Remote     spots    bases      avgLength size_MB
247 MBJMR04201 SRR6323266 1    SRR6323266 12442511 3344951562 268       1447   
242 LHCFC01400 SRR6323278 1    SRR6323278 15734759 4534806422 288       1814   
73  DHKFI23503 SRR6323520 1    SRR6323520 16033611 4697664620 292       1703   
    BioProject  bioproject  ⋯
247 PRJNA417359 PRJNA417359 ⋯
242 PRJNA417359 PRJNA417359 ⋯
73  PRJNA417359 PRJNA417359 ⋯
    Name                                                                                 
247 Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
242 Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
73  Agricultural Genomics Institute at Shenzhen, Chinese Academy of Agricultural Sciences
    SRA_accession
247 NA           
242 NA           
73  NA           
    Title                                                          Animal 
247 Metagenome or environmental sample from chicken gut metagenome Chicken
242 Metagenome or environmental sample from chicken gut metagenome Chicken
73  Metagenome or environmental sample from chicken gut metagenome Chicken
    First_author Year DOI                                      
247 Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
242 Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
73  Huang        2018 https://doi.org/10.1186/s40168-018-0590-5
    Sequencer           Read_length Comments
247 HiSeq2500, HiSeqX10 NA          NA      
242 HiSeq2500, HiSeqX10 NA          NA      
73  HiSeq2500, HiSeqX10 NA          NA

In [38]:
samps %>% 
    dplyr::select(spots, bases, avgLength, size_MB) %>%
    summary

     spots              bases             avgLength        size_MB    
 Min.   :10048833   Min.   :2.588e+09   Min.   :237.0   Min.   : 972  
 1st Qu.:12347516   1st Qu.:3.227e+09   1st Qu.:246.0   1st Qu.:1302  
 Median :15603560   Median :4.162e+09   Median :286.5   Median :1562  
 Mean   :15487376   Mean   :4.241e+09   Mean   :273.2   Mean   :1611  
 3rd Qu.:17945110   3rd Qu.:4.851e+09   3rd Qu.:289.8   3rd Qu.:1840  
 Max.   :33497733   Max.   :9.693e+09   Max.   :294.0   Max.   :3848  

In [40]:
F = file.path(work_dir, projectID, 'samples_n130.txt')
write.table(samps, file=F, sep='\t', quote=FALSE, row.names=FALSE)
cat('File written:', F, '\n')

File written: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA417359/samples_n130.txt 


### Config

In [41]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA417359/samples_n130.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA417359/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for r

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/vadinCA11/bin/llmgqc
$ screen -L -S llmgqc-PRJNA417359 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA417359/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA417359/LLMGQC/ 20
```

### Summary

In [42]:
projectID = 'PRJNA417359'

In [43]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 260  21


Sample                                    Filename      
1 CBJMC01400 | fastqc_per_sample | R1_final R1_final.fq.gz
2 CBJMC01400 | fastqc_per_sample | R2_final R2_final.fq.gz
3 CBJMI02800 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9  9720627       
2 Conventional base calls Sanger / Illumina 1.9  9720627       
3 Conventional base calls Sanger / Illumina 1.9 10246817       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-150          52  
2 0                                 90-150          52  
3 0                                 90-150          46  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 83.80389                      148.1134            ⋯ pass                     
2 84.30671                      132.3763            ⋯ pass                     
3 56.69880                      148.5616            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 fail                      pass                       
2 fail                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 pass                      warn                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         warn                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [44]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   : 8968487   Min.   :118.6      
 1st Qu.:11767226   1st Qu.:124.1      
 Median :15002040   Median :148.0      
 Mean   :14763142   Mean   :140.0      
 3rd Qu.:17069678   3rd Qu.:148.4      
 Max.   :32055696   Max.   :149.0      

## PRJNA476660

* host = Pangolin
* n = 3
* avg. read length = 300

In [45]:
projectID = 'PRJNA476660'

In [47]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Directory already exists: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA476660/LLMGQC 


In [ ]:
cat_file(file.path(outdir, 'config.yaml'))

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/vadinCA11/bin/llmgqc
$ screen -L -S llmgqc-PRJNA476660 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA476660/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA476660/LLMGQC/SGE_log/ 20
```

### Summary

In [53]:
projectID = 'PRJNA476660'

In [49]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1]  6 20


Sample                                Filename       File.type              
1 MJGMD1 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
2 MJGMD1 | fastqc_per_sample | R2_final R2_final.fq.gz Conventional base calls
3 MJGMH1 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
  Encoding              Total.Sequences Sequences.flagged.as.poor.quality
1 Sanger / Illumina 1.9 24099125        0                                
2 Sanger / Illumina 1.9 24099125        0                                
3 Sanger / Illumina 1.9 23757188        0                                
  Sequence.length X.GC total_deduplicated_percentage avg_sequence_length
1 90-150          46   81.90689                      149.6247           
2 90-150          46   83.74366                      149.4667           
3 90-150          44   76.84584                      149.6505           
  basic_statistics per_base_sequence_quality per_sequence_quality_scores
1 pass             pass                      pass                       
2 pass             pass                      pass                       
3 pass             pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 pass                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [50]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   :23515478   Min.   :149.6      
 1st Qu.:23636333   1st Qu.:149.6      
 Median :23757188   Median :149.6      
 Mean   :23790597   Mean   :149.6      
 3rd Qu.:23928156   3rd Qu.:149.6      
 Max.   :24099125   Max.   :149.7      

## PRJNA485217

* host = primate (capuchin)
* n = 19
* avg. read length = 250

In [51]:
projectID = 'PRJNA485217'

In [52]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA485217/LLMGQC 


In [54]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA485217/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA485217/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/vadinCA11/bin/llmgqc
$ screen -L -S llmgqc-PRJNA485217 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA485217/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA485217/LLMGQC/SGE_log/ 20
```

### Summary

In [55]:
projectID = 'PRJNA485217'

In [56]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 38 20


Sample                                                                    
1 Cebus_capucinus_imitator_PRJNA485217_SSR007 | fastqc_per_sample | R1_final
2 Cebus_capucinus_imitator_PRJNA485217_SSR007 | fastqc_per_sample | R2_final
3 Cebus_capucinus_imitator_PRJNA485217_SSR012 | fastqc_per_sample | R1_final
  Filename       File.type               Encoding              Total.Sequences
1 R1_final.fq.gz Conventional base calls Sanger / Illumina 1.9 6307466        
2 R2_final.fq.gz Conventional base calls Sanger / Illumina 1.9 6307466        
3 R1_final.fq.gz Conventional base calls Sanger / Illumina 1.9 5481860        
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-126          52  
2 0                                 90-126          52  
3 0                                 90-126          49  
  total_deduplicated_percentage avg_sequence_length basic_statistics
1 87.46016                      125.3802            pass            
2 87.75630                      125.2676            pass            
3 88.82229                      123.3990            pass            
  per_base_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 warn                      fail                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [57]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences   avg_sequence_length
 Min.   :4385654   Min.   :123.4      
 1st Qu.:5573106   1st Qu.:124.6      
 Median :6387158   Median :125.0      
 Mean   :6344394   Mean   :124.8      
 3rd Qu.:6734097   3rd Qu.:125.0      
 Max.   :8008710   Max.   :125.4      

## PRJNA532626

* host = Black rhinoceros
* n = 25
* avg. read length = 290

In [58]:
projectID = 'PRJNA532626'

In [60]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Directory already exists: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJNA532626/LLMGQC 


In [61]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA532626/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA532626/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/vadinCA11/bin/llmgqc
$ screen -L -S llmgqc-PRJNA532626 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA532626/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJNA532626/LLMGQC/SGE_log/ 20
```

### Summary

In [62]:
projectID = 'PRJNA532626'

In [64]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 50 20


Sample                             Filename       File.type              
1 R01 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
2 R01 | fastqc_per_sample | R2_final R2_final.fq.gz Conventional base calls
3 R02 | fastqc_per_sample | R1_final R1_final.fq.gz Conventional base calls
  Encoding              Total.Sequences Sequences.flagged.as.poor.quality
1 Sanger / Illumina 1.9 16152672        0                                
2 Sanger / Illumina 1.9 16152672        0                                
3 Sanger / Illumina 1.9 18542644        0                                
  Sequence.length X.GC total_deduplicated_percentage avg_sequence_length
1 90-150          48   92.32873                      147.4574           
2 90-150          48   92.32873                      147.4574           
3 90-150          46   89.26856                      147.4870           
  basic_statistics per_base_sequence_quality per_sequence_quality_scores
1 pass             pass                      pass                       
2 pass             pass                      pass                       
3 pass             pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 fail                      pass                    pass              
2 fail                      pass                    pass              
3 fail                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [65]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   : 6782502   Min.   :141.9      
 1st Qu.: 9975355   1st Qu.:146.3      
 Median :11410826   Median :146.9      
 Mean   :12299509   Mean   :146.4      
 3rd Qu.:14608744   3rd Qu.:147.3      
 Max.   :18565738   Max.   :147.5      

## PRJEB22765

* host = Macaque (Macaca fascicularis)
* n = 20

In [69]:
projectID = 'PRJEB22765'

In [70]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB22765/LLMGQC 


In [72]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB22765/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB22765/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote (

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJEB22765 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB22765/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB22765/LLMGQC/SGE_log/ 20
```

### Summary

In [73]:
projectID = 'PRJEB22765'

In [74]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 40 21


Sample                                        Filename      
1 SAMEA104315420 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA104315420 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA104315421 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 29900443       
2 Conventional base calls Sanger / Illumina 1.9 29900443       
3 Conventional base calls Sanger / Illumina 1.9 27198790       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-95           47  
2 0                                 90-95           47  
3 0                                 90-95           46  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 93.50983                      94.86005            ⋯ pass                     
2 93.68126                      94.81615            ⋯ pass                     
3 95.65863                      94.87778            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 warn                      pass                       
2 warn                      pass                       
3 warn                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 pass                      pass                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [75]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   :21314113   Min.   :90.97      
 1st Qu.:24912732   1st Qu.:94.88      
 Median :27186220   Median :94.88      
 Mean   :27934239   Mean   :94.71      
 3rd Qu.:28695570   3rd Qu.:94.95      
 Max.   :38025452   Max.   :96.91      

## PRJEB23642

* host = Minke whale
* n = 12

In [76]:
projectID = 'PRJEB23642'

In [77]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB23642/LLMGQC 


In [78]:
cat_file(file.path(outdir, 'config.yaml'))

#-- I/O --#
# table with sample --> read_file information
samples_file: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB23642/samples.txt

# output location
output_dir: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB23642/LLMGQC/

# read file path
# use "None" if full file path is included in the samples_file
read_file_path: None

#-- Software parameters --#
# Use "Skip" to skip any of these steps. If no params for rule, use ""
# `clumpify`: change dupedist if not HiSeq3000/4000 (dupedist=40 for NextSeq, HiSeq2500, and MiSeq)
params:
  # read download
  remote: just_single=False      # If True, just download read1 (if remote file)
  # validation, conversion, subsampling
  validate_reads: ""
  convert_fastq_to_1.8: ""
  seqtk_sample: Skip    # Use number to subsample reads (eg., 1000000)
  fastqc_on_raw: ""
  # de-duplication
  clumpify: Skip #dedupe=t dupedist=2500 optical=t    # this will likely fail for remote (

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJEB23642 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB23642/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB23642/LLMGQC/SGE_log/ 20
```

### Summary

In [84]:
projectID = 'PRJEB23642'

In [85]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 24 21


Sample                                        Filename      
1 SAMEA104404095 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA104404095 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA104404096 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 30215878       
2 Conventional base calls Sanger / Illumina 1.9 30215878       
3 Conventional base calls Sanger / Illumina 1.9 35937268       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-251          45  
2 0                                 90-251          45  
3 0                                 90-251          41  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 78.77218                      249.2120            ⋯ pass                     
2 80.38767                      247.2523            ⋯ pass                     
3 84.58096                      249.6105            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 pass                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 pass                      pass                    pass              
2 pass                      pass                    pass              
3 pass                      warn                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         pass                       
2 warn                         pass                       
3 warn                         pass                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [86]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   : 7815889   Min.   :229.0      
 1st Qu.:23235914   1st Qu.:249.2      
 Median :24528324   Median :249.4      
 Mean   :24451255   Mean   :247.8      
 3rd Qu.:28808351   3rd Qu.:249.7      
 Max.   :35937268   Max.   :250.1      

## PRJEB9357

* host = house cat
* n = 87

In [87]:
projectID = 'PRJEB9357'

In [88]:
outdir = file.path(work_dir, projectID, 'LLMGQC')
make_dir(outdir)

Created directory: /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects//PRJEB9357/LLMGQC 


In [ ]:
cat_file(file.path(outdir, 'config.yaml'))

### Run

```
(snakemake_dev) @ rick:/ebio/abt3_projects/methanogen_host_evo/bin/llmgqc
$ screen -L -S llmgqc-PRJEB9357 ./snakemake_sge.sh /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB9357/LLMGQC/config.yaml cluster.json /ebio/abt3_projects/Georg_animal_feces/data/metagenome/multi-study/BioProjects/PRJEB9357/LLMGQC/SGE_log/ 20
```

### Summary

In [89]:
projectID = 'PRJEB9357'

In [90]:
F = file.path(work_dir, projectID, 'LLMGQC', 'reports', 'multiqc-final', 'multiqc_data', 'multiqc_fastqc.txt')
fastqc = read.delim(F, sep='\t')
fastqc %>% dfhead

[1] 174  21


Sample                                      Filename      
1 SAMEA3387433 | fastqc_per_sample | R1_final R1_final.fq.gz
2 SAMEA3387433 | fastqc_per_sample | R2_final R2_final.fq.gz
3 SAMEA3387434 | fastqc_per_sample | R1_final R1_final.fq.gz
  File.type               Encoding              Total.Sequences
1 Conventional base calls Sanger / Illumina 1.9 14032468       
2 Conventional base calls Sanger / Illumina 1.9 14032468       
3 Conventional base calls Sanger / Illumina 1.9 18497450       
  Sequences.flagged.as.poor.quality Sequence.length X.GC
1 0                                 90-100          48  
2 0                                 90-100          48  
3 0                                 90-100          47  
  total_deduplicated_percentage avg_sequence_length ⋯ per_base_sequence_quality
1 69.02745                      99.46529            ⋯ pass                     
2 70.74038                      99.37996            ⋯ pass                     
3 64.07215                      99.47550            ⋯ pass                     
  per_tile_sequence_quality per_sequence_quality_scores
1 pass                      pass                       
2 fail                      pass                       
3 pass                      pass                       
  per_base_sequence_content per_sequence_gc_content per_base_n_content
1 fail                      fail                    pass              
2 fail                      fail                    pass              
3 fail                      warn                    pass              
  sequence_length_distribution sequence_duplication_levels
1 warn                         warn                       
2 warn                         pass                       
3 warn                         warn                       
  overrepresented_sequences adapter_content
1 pass                      pass           
2 pass                      pass           
3 pass                      pass

In [91]:
fastqc %>% 
    filter(Filename == 'R1_final.fq.gz') %>%
    dplyr::select(Total.Sequences, avg_sequence_length) %>% 
    summary

 Total.Sequences    avg_sequence_length
 Min.   : 7783348   Min.   :99.29      
 1st Qu.:16676332   1st Qu.:99.43      
 Median :19016059   Median :99.48      
 Mean   :19235981   Mean   :99.49      
 3rd Qu.:20928948   3rd Qu.:99.50      
 Max.   :40370269   Max.   :99.85      

# sessionInfo

In [36]:
sessionInfo()

R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS: /ebio/abt3_projects/software/miniconda3_gt4.4/envs/py3_physeq/lib/R/lib/libRblas.so
LAPACK: /ebio/abt3_projects/software/miniconda3_gt4.4/envs/py3_physeq/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2 ggplot2_3.0.0  tidyr_0.8.1    dplyr_0.7.6   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.17     bindr_0.1.1      magrittr_1.5     munsell_0.5.0   
 [5] tidyselect_0.2